app/assets/javascripts/discourse/app/static/wizard/components/wizard-step.gjs

```diff
@@ -0,0 +1,302 @@
+              >
+                {{i18n "wizard.next"}}
+              </button>
+            {{/if}}
+
+          </div>
+
+        </div>
+      </div>
+    </div>
+  </template>
+
+  @tracked saving = false;
+
+  get wizard() {
+    return this.args.wizard;
+  }
+
+  get step() {
+    return this.args.step;
+  }
+
+  get id() {
+    return this.step.id;
+  }
+
+  /**
+    Step        Back Button?     Primary Action      Secondary Action
+    ------------------------------------------------------------------
+    First            No               Next                  N/A
+    ------------------------------------------------------------------
+    ...             Yes               Next                  N/A
+    ------------------------------------------------------------------
+    Ready           Yes              Jump In          Configure More
+    ------------------------------------------------------------------
+    ...             Yes               Next              Exit Setup
+    ------------------------------------------------------------------
+    Last            Yes              Jump In                N/A
+    ------------------------------------------------------------------
+
+    Back Button: without saving, go back to the last page
+    Next Button: save, and if successful, go to the next page
+    Configure More: re-skinned next button
+    Exit Setup: without saving, go to the home page ("finish")
+    Jump In: on the "ready" page, it exits the setup ("finish"), on the
+    last page, it saves, and if successful, go to the home page
+
+   */
+  get isFinalStep() {
+    return this.step.displayIndex === this.wizard.steps.length;
+  }
+
+  get showBackButton() {
+    return this.step.index > 0;
+  }
+
+  get showFinishButton() {
+    const ready = this.wizard.findStep("ready");
+    const isReady = ready && this.step.index > ready.index;
+    return isReady && !this.isFinalStep;
+  }
+
+  get showConfigureMore() {
+    return this.id === "ready";
+  }
+
+  get showJumpInButton() {
+    return this.id === "ready" || this.isFinalStep;
+  }
+
+  get includeSidebar() {
+    return !!this.step.fields.find((f) => f.showInSidebar);
+  }
+
+  @action
+  stepChanged() {
+    this.saving = false;
+    this.autoFocus();
+  }
+
+  @action
+  onKeyUp(event) {
+    if (event.key === "Enter") {
+      if (this.showJumpInButton) {
+        this.jumpIn();
+      } else {
+        this.nextStep();
+      }
+    }
+  }
+
+  @action
+  autoFocus() {
+    schedule("afterRender", () => {
+      const $invalid = $(
+        ".wizard-container__input.invalid:nth-of-type(1) .wizard-focusable"
+      );
+
+      if ($invalid.length) {
+        return $invalid.focus();
+      }
+
+      $(".wizard-focusable:nth-of-type(1)").focus();
+    });
+  }
+
+  async advance() {
+    try {
+      this.saving = true;
+      const response = await this.step.save();
+      this.args.goNext(response);
+    } finally {
+      this.saving = false;
+    }
+  }
```

Comment:
This doesn't exactly do what you would imagine (but is ported over exactly as-is). You would think the try/catch is there to guard the `await this.step.save()` step failing, but that step actually never fails, because it has a `catch` that never re-throws the error.

What actually happens is that if an error happens in `this.step.save()`, it will fail to return a response value (`response === undefined`), and the code in `@goNext` is guarded with `response?.blah` such that it doesn't actually do anything if response is undefined.

I don't think that is how anyone is envisioning things to work 😬 

Probably should make `step.save` re-throw, so that we never call `this.args.goNext` when the save fails. But I left it for someone else to tackle.